src/nvim/ex_cmds.c

```diff
@@ -1583,22 +1583,43 @@ char *make_filter_cmd(char *cmd, char *itmp, char *otmp)
   if (otmp != NULL) {
     len += STRLEN(otmp) + STRLEN(p_srr) + 2;  // two extra spaces ("  "),
   }
+
+  const char *const cmd_args = strchr(cmd, ' ');
+  len += (is_pwsh && cmd_args)
+      ? STRLEN(" -ArgumentList ") + 2  // two extra quotes
+      : 0;
+
   char *const buf = xmalloc(len);
 
-#if defined(UNIX)
-  // Put delimiters around the command (for concatenated commands) when
-  // redirecting input and/or output.
   if (is_pwsh) {
     xstrlcpy(buf, "Start-Process ", len);
-    xstrlcat(buf, cmd, len);
+    if (cmd_args == NULL) {
+      xstrlcat(buf, cmd, len);
+    } else {
+      char *con = NULL;
+      char *dupcmd = xstrdup(cmd);
+      const char *const cmd_cmd = os_strtok(dupcmd, " ", &con);
+      xstrlcat(buf, cmd_cmd, len);
```

Comment:
`cmd_cmd` may be NULL if `dupcmd` does not contains `" "`.

Reply:
I handled that case with `cmd_args`. If `cmd` (and by extension `dupcmd`) doesn't contain a space, then `cmd_args` will be NULL, so this block won't be executed.

Reply:
But maybe I should add an if(!null) anyway, to be idiomatic.